--- a/dmenu.c	2021-06-08 16:30:56.008937367 +0300
+++ b/dmenu.c	2021-06-08 17:09:17.179790158 +0300
@@ -352,6 +352,7 @@ keypress(XKeyEvent *ev)
 		switch(ksym) {
 		case XK_a: ksym = XK_Home;      break;
 		case XK_b: ksym = XK_Left;      break;
+		case XK_l: ksym = XK_Escape;    break;
 		case XK_c: ksym = XK_Escape;    break;
 		case XK_d: ksym = XK_Delete;    break;
 		case XK_e: ksym = XK_End;       break;
@@ -401,18 +401,25 @@ keypress(XKeyEvent *ev)
 		}
 	} else if (ev->state & Mod1Mask) {
 		switch(ksym) {
-		case XK_b:
+		case XK_h:
 			movewordedge(-1);
 			goto draw;
-		case XK_f:
+		case XK_i:
 			movewordedge(+1);
 			goto draw;
-		case XK_g: ksym = XK_Home;  break;
-		case XK_G: ksym = XK_End;   break;
-		case XK_h: ksym = XK_Up;    break;
+		case XK_n: ksym = XK_Down;  break;
+		case XK_e: ksym = XK_Up;    break;
 		case XK_j: ksym = XK_Next;  break;
 		case XK_k: ksym = XK_Prior; break;
-		case XK_l: ksym = XK_Down;  break;
+		case XK_l: /* delete left */
+			insert(NULL, 0 - cursor);
+			break;
+		case XK_y: /* delete right */
+			text[cursor] = '\0';
+			match();
+			break;
+		case XK_g: ksym = XK_Home;  break;
+		case XK_G: ksym = XK_End;   break;
 		default:
 			return;
 		}
